client/src/views/users/UpdatePasswordModal.vue

```diff
@@ -0,0 +1,203 @@
+          >
+            {{ getNextButtonText() }}
+          </ion-button>
+        </ion-buttons>
+      </ion-footer>
+    </div>
+  </ion-page>
+</template>
+
+<script setup lang="ts">
+import { asyncComputed } from '@/common/asyncComputed';
+import { MsChoosePasswordInput } from '@/components/core';
+import MsPasswordInput from '@/components/core/ms-input/MsPasswordInput.vue';
+import { MsModalResult } from '@/components/core/ms-modal/types';
+import { AvailableDevice, ChangeAuthErrorTag, changePassword as parsecChangePassword } from '@/parsec';
+import { Information, InformationKey, InformationLevel, InformationManager, PresentationMode } from '@/services/informationManager';
+import { translate } from '@/services/translation';
+import { IonButton, IonButtons, IonFooter, IonHeader, IonIcon, IonPage, IonTitle, modalController } from '@ionic/vue';
+import { close } from 'ionicons/icons';
+import { Ref, inject, onMounted, ref } from 'vue';
+
+enum ChangePasswordStep {
+  OldPassword,
+  NewPassword,
+}
+
+const pageStep = ref(ChangePasswordStep.OldPassword);
+const currentDevice: Ref<AvailableDevice> = ref({} as AvailableDevice);
+const choosePasswordInput: Ref<typeof MsChoosePasswordInput | null> = ref(null);
+const informationManager: InformationManager = inject(InformationKey)!;
+const fieldsUpdated = ref(false);
+const oldPassword = ref('');
+const errorMessage = ref('');
+const passwordIsInvalid = ref(false);
+const currentPasswordInput = ref();
+
+onMounted(() => {
+  currentPasswordInput.value.setFocus();
+});
+
+const changeButtonIsEnabled = asyncComputed(async (): Promise<boolean> => {
+  // forces the update
+  fieldsUpdated.value;
+  return choosePasswordInput.value && (await choosePasswordInput.value.areFieldsCorrect()) && oldPassword.value.length > 0;
+});
+
+async function nextStep(): Promise<void> {
+  if (pageStep.value === ChangePasswordStep.OldPassword) {
+    await checkCurrentPassword();
+  } else if (pageStep.value === ChangePasswordStep.NewPassword) {
+    await changePassword();
+  }
+}
+
+async function checkCurrentPassword(): Promise<boolean | undefined> {
+  if (oldPassword.value !== 'P@ssw0rd.') {
+    errorMessage.value = translate('HomePage.organizationLogin.passwordError');
+    passwordIsInvalid.value = true;
+    return false;
+  }
+  pageStep.value = ChangePasswordStep.NewPassword;
+}
+
+async function changePassword(): Promise<void> {
+  if (!(await choosePasswordInput.value?.areFieldsCorrect())) {
+    console.log('Password is not correct');
+    return;
+  }
+
+  const result = await parsecChangePassword(currentDevice.value, oldPassword.value, choosePasswordInput.value?.password);
+
+  if (result.ok) {
+    informationManager.present(
+      new Information({
+        message: translate('ContactDetailsPage.passwordUpdated'),
+        level: InformationLevel.Success,
+      }),
+      PresentationMode.Toast,
+    );
```

Comment:
informationManager.present is asynchrone, maybe you should add an await here?

Reply:
For a toast it should be fine.